lib/python/reduction.cpp

```diff
@@ -18,14 +18,17 @@ template <class T> void bind_mean(py::module &m) {
       "mean", [](const T &x) { return mean(x); }, py::arg("x"),
       py::call_guard<py::gil_scoped_release>());
   m.def(
-      "mean", [](const T &x, const Dim dim) { return mean(x, dim); },
+      "mean",
+      [](const T &x, const std::string &dim) { return mean(x, Dim{dim}); },
```

Comment:
Question, without having looked at everything: Many functions such as these are anyway wrapped on the Python side. Can we keep `Dim` for internal use here?

Reply:
Yes, that would be possible. But if we want scipp to pass type checks, we would have to pass `Dim(dim_str)` as arguments from Python.

Or maybe we can modify the type hints to be `Union[Dim, str]`. Gonna look into this. But the public API should not reference `Dim` at all, I think.